tracpro/contacts/models.py

```diff
@@ -97,10 +100,12 @@ def kwargs_from_temba(cls, org, temba_contact):
 
         org_region_uuids = [r.uuid for r in Region.get_all(org)]
         region_uuids = intersection(org_region_uuids, temba_contact.groups)
-        region = Region.objects.get(org=org, uuid=region_uuids[0]) if region_uuids else None
-
-        if not region:  # pragma: no cover
-            raise ValueError("No region with UUID in %s" % ", ".join(temba_contact.groups))
+        if len(region_uuids) > 0:
+            region = Region.objects.filter(org=org, uuid__in=region_uuids).first() if region_uuids else None
+            if not region:  # pragma: no cover
```

Comment:
No I think a contact should have a region, but if you want to know which contact is failing - just add the contact id to the original error message. I'm pretty sure that with this change, the exception on line 106 will never be raised.